src/instana/instrumentation/pika.py

```diff
@@ -2,167 +2,244 @@
 # (c) Copyright IBM Corp. 2021
 # (c) Copyright Instana Inc. 2021
 
+import types
+from typing import TYPE_CHECKING, Any, Callable, Dict, Generator, Optional, Tuple, Union
 
-
+import pika.connection
 import wrapt
-import opentracing
-import types
 
-from ..log import logger
-from ..singletons import tracer
-from ..util.traceutils import get_tracer_tuple, tracing_is_off
+from instana.log import logger
+from instana.propagators.format import Format
+from instana.singletons import tracer
+from instana.util.traceutils import get_tracer_tuple, tracing_is_off
+
+if TYPE_CHECKING:
+    from instana.span.span import InstanaSpan
 
 try:
     import pika
 
-
-    def _extract_broker_tags(span, conn):
-        span.set_tag("address", "%s:%d" % (conn.params.host, conn.params.port))
-
-
-    def _extract_publisher_tags(span, conn, exchange, routing_key):
+    if TYPE_CHECKING:
+        import pika.adapters.blocking_connection
+        import pika.channel
+
+    def _extract_broker_tags(
+        span: "InstanaSpan", conn: pika.connection.Connection
+    ) -> None:
+        span.set_attribute("address", f"{conn.params.host}:{conn.params.port}")
+
+    def _extract_publisher_tags(
```

Comment:
Similar to https://github.com/instana/python-sensor/pull/614/files#r1762368639

Reply:
Done